Modules/Tracker/QuestieTrackerPrivates.lua

```diff
@@ -8,56 +8,335 @@ local QuestieCombatQueue = QuestieLoader:ImportModule("QuestieCombatQueue")
+            elseif baseFrame:GetHeight()/GetScreenHeight() > 0.25 and (Questie.db.char.TrackerLocation[1] == "BOTTOMLEFT" or Questie.db.char.TrackerLocation[1] == "BOTTOMRIGHT") then
+                yAdj = -baseFrame:GetHeight()/2
+            elseif baseFrame:GetHeight()/GetScreenHeight() > 0.75 then
+                yAdj = baseFrame:GetHeight()
+            end
+
+            -- setPoint Topleft = Down and Right
+            if Questie.db.char.TrackerLocation[4] < 0 and Questie.db.char.TrackerLocation[5] > yAdj and Questie.db.char.TrackerLocation[1] ~= "TOPLEFT" then
+                xOff, yOff = _QuestieTracker:ConvertSetPointCords(baseFrame, "TOPLEFT")
+                Questie.db.char.TrackerLocation[1] = "TOPLEFT"
+                Questie.db.char.TrackerLocation[4] = xOff
+                Questie.db.char.TrackerLocation[5] = yOff
+
+            -- setPoint Topright = Down and Left
+            elseif Questie.db.char.TrackerLocation[4] > 0 and Questie.db.char.TrackerLocation[5] > yAdj and Questie.db.char.TrackerLocation[1] ~= "TOPRIGHT" then
+                xOff, yOff = _QuestieTracker:ConvertSetPointCords(baseFrame, "TOPRIGHT")
+                Questie.db.char.TrackerLocation[1] = "TOPRIGHT"
+                Questie.db.char.TrackerLocation[4] = xOff
+                Questie.db.char.TrackerLocation[5] = yOff
+
+            -- setPoint Bottomleft = Up and Right
+            elseif Questie.db.char.TrackerLocation[4] < 0 and Questie.db.char.TrackerLocation[5] < yAdj and Questie.db.char.TrackerLocation[1] ~= "BOTTOMLEFT" then
+                xOff, yOff = _QuestieTracker:ConvertSetPointCords(baseFrame, "BOTTOMLEFT")
+                Questie.db.char.TrackerLocation[1] = "BOTTOMLEFT"
+                Questie.db.char.TrackerLocation[4] = xOff
+                Questie.db.char.TrackerLocation[5] = yOff
+
+            -- setPoint Bottomright = Up and Left
+            elseif Questie.db.char.TrackerLocation[4] > 0 and Questie.db.char.TrackerLocation[5] < yAdj and Questie.db.char.TrackerLocation[1] ~= "BOTTOMRIGHT" then
+                xOff, yOff = _QuestieTracker:ConvertSetPointCords(baseFrame, "BOTTOMRIGHT")
+                Questie.db.char.TrackerLocation[1] = "BOTTOMRIGHT"
+                Questie.db.char.TrackerLocation[4] = xOff
+                Questie.db.char.TrackerLocation[5] = yOff
+            end
+
+            Questie.db.char.TrackerLocation[2] = "UIParent"
+            Questie.db.char.TrackerLocation[3] = "CENTER"
+
+            baseFrame:ClearAllPoints()
+            baseFrame:SetPoint(unpack(Questie.db.char.TrackerLocation))
+
+        -- When the user sets a manual setPoint it runs QuestieTracker:ResetLocation() which
+        -- converts everything over to a [CUSTOM]/UIParent/CENTER centric x,y cords so no pre/post
+        -- processing of location data is needed. It's done "as-is".
+        else
+            Questie.db.char.TrackerLocation[1] = Questie.db.char.trackerSetpoint
+        end
```

Comment:
This should be refactored to be multiple functions

Reply:
Fixed in latest commit